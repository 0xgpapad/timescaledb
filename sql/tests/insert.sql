\set ON_ERROR_STOP 1

\ir create_clustered_db.sql

\set ECHO ALL
\c meta
SELECT add_cluster_user('postgres', NULL);

SELECT add_node('Test1' :: NAME, 'localhost');
SELECT add_node('test2' :: NAME, 'localhost');

SELECT add_namespace('testNs' :: NAME);
SELECT add_field('testNs' :: NAME, 'device_id', 'text', TRUE, TRUE, ARRAY ['VALUE-TIME'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_0', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_1', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_2', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_bool', 'boolean', FALSE, FALSE, ARRAY ['TIME-VALUE'] :: field_index_type []);

\c Test1
BEGIN;
SELECT *
FROM create_temp_copy_table_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
\COPY copy_t FROM 'data/ds1_dev1_1.tsv';
SELECT *
FROM insert_data_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
COMMIT;

SELECT close_data_table_end(dt.table_oid)
FROM data_table dt
WHERE dt.namespace_name = 'testNs';
BEGIN;
SELECT *
FROM create_temp_copy_table_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
\COPY copy_t FROM 'data/ds1_dev1_2.tsv';
SELECT *
FROM insert_data_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
COMMIT;

\c test2

BEGIN;
SELECT *
FROM create_temp_copy_table_one_partition('copy_t', get_partition_for_key('dev2', 10 :: SMALLINT), 10 :: SMALLINT);
\COPY copy_t FROM 'data/ds1_dev2_1.tsv';
SELECT *
FROM insert_data_one_partition('copy_t', get_partition_for_key('dev2', 10 :: SMALLINT), 10 :: SMALLINT);
COMMIT;

\c Test1
\dt "testNs".*

\c test2
\dt "testNs".*
SELECT *
FROM "testNs".cluster;
SELECT *
FROM "testNs".distinct;

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs'));
SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate=> new_aggregate(10e9::BIGINT)
                     ));
