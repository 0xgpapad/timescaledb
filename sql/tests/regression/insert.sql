\set ON_ERROR_STOP 1

\ir create_clustered_db.sql

\set ECHO ALL
\c meta
SELECT add_cluster_user('postgres', NULL);

SELECT set_meta('meta' :: NAME, 'localhost');
SELECT add_node('Test1' :: NAME, 'localhost');
SELECT add_node('test2' :: NAME, 'localhost');

SELECT add_hypertable('testNs' :: NAME, 'device_id', 'testNs', 'testNs' );
SELECT add_field('testNs' :: NAME, 'device_id', 'text', TRUE, TRUE, ARRAY ['VALUE-TIME'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_0', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_1', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_2', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
SELECT add_field('testNs' :: NAME, 'series_bool', 'boolean', FALSE, FALSE, ARRAY ['TIME-VALUE'] :: field_index_type []);

\c Test1
BEGIN;
SELECT *
FROM create_temp_copy_table('copy_t');
\COPY copy_t FROM 'data/ds1_dev1_1.tsv';
SELECT *
FROM insert_data('copy_t');
COMMIT;

SELECT close_chunk_end(c.id)
FROM get_open_partition_for_key('testNs','dev1') part
INNER JOIN chunk c ON (c.partition_id = part.id);

\c Test1
BEGIN;
SELECT *
FROM create_temp_copy_table('copy_t');
\COPY copy_t FROM 'data/ds1_dev1_2.tsv';
SELECT *
FROM insert_data('copy_t');
COMMIT;

\c test2
BEGIN;
SELECT *
FROM create_temp_copy_table('copy_t');
\COPY copy_t FROM 'data/ds1_dev2_1.tsv';
SELECT *
FROM insert_data('copy_t');
COMMIT;

\c Test1
\d+ "testNs".*

\c test2
\d+ "testNs".*
SELECT *
FROM "testNs"._hyper_1_0_replica;
SELECT *
FROM "testNs"._hyper_1_0_distinct;


