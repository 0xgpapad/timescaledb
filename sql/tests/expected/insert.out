\c meta
SELECT add_cluster_user('postgres', NULL);
 add_cluster_user 
------------------
 
(1 row)

SELECT add_node('Test1' :: NAME, 'localhost');
 add_node 
----------
 
(1 row)

SELECT add_node('test2' :: NAME, 'localhost');
 add_node 
----------
 
(1 row)

SELECT add_namespace('testNs' :: NAME);
 add_namespace 
---------------
 
(1 row)

SELECT add_field('testNs' :: NAME, 'device_id', 'text', TRUE, TRUE, ARRAY ['VALUE-TIME'] :: field_index_type []);
 add_field 
-----------
 
(1 row)

SELECT add_field('testNs' :: NAME, 'series_0', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
 add_field 
-----------
 
(1 row)

SELECT add_field('testNs' :: NAME, 'series_1', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
 add_field 
-----------
 
(1 row)

SELECT add_field('testNs' :: NAME, 'series_2', 'double precision', FALSE, FALSE,
                 ARRAY ['TIME-VALUE'] :: field_index_type []);
 add_field 
-----------
 
(1 row)

SELECT add_field('testNs' :: NAME, 'series_bool', 'boolean', FALSE, FALSE, ARRAY ['TIME-VALUE'] :: field_index_type []);
 add_field 
-----------
 
(1 row)

\c Test1
BEGIN;
SELECT *
FROM create_temp_copy_table_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
 create_temp_copy_table_one_partition 
--------------------------------------
 copy_t
(1 row)

\COPY copy_t FROM 'data/ds1_dev1_1.tsv';
SELECT *
FROM insert_data_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
 insert_data_one_partition 
---------------------------
 
(1 row)

COMMIT;
SELECT close_data_table_end(dt.table_oid)
FROM data_table dt
WHERE dt.namespace_name = 'testNs';
 close_data_table_end 
----------------------
 
(1 row)

BEGIN;
SELECT *
FROM create_temp_copy_table_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
 create_temp_copy_table_one_partition 
--------------------------------------
 copy_t
(1 row)

\COPY copy_t FROM 'data/ds1_dev1_2.tsv';
SELECT *
FROM insert_data_one_partition('copy_t', get_partition_for_key('dev1', 10 :: SMALLINT), 10 :: SMALLINT);
 insert_data_one_partition 
---------------------------
 
(1 row)

COMMIT;
\c test2
BEGIN;
SELECT *
FROM create_temp_copy_table_one_partition('copy_t', get_partition_for_key('dev2', 10 :: SMALLINT), 10 :: SMALLINT);
 create_temp_copy_table_one_partition 
--------------------------------------
 copy_t
(1 row)

\COPY copy_t FROM 'data/ds1_dev2_1.tsv';
SELECT *
FROM insert_data_one_partition('copy_t', get_partition_for_key('dev2', 10 :: SMALLINT), 10 :: SMALLINT);
 insert_data_one_partition 
---------------------------
 
(1 row)

COMMIT;
\c Test1
\dt "testNs".*
                List of relations
 Schema |         Name         | Type  |  Owner   
--------+----------------------+-------+----------
 testNs | cluster              | table | postgres
 testNs | data_4_10_1257811200 | table | postgres
 testNs | data_4_10_1257984000 | table | postgres
 testNs | distinct             | table | postgres
 testNs | local_distinct       | table | postgres
 testNs | master               | table | postgres
 testNs | partition_4_10       | table | postgres
(7 rows)

\c test2
\dt "testNs".*
                List of relations
 Schema |         Name         | Type  |  Owner   
--------+----------------------+-------+----------
 testNs | cluster              | table | postgres
 testNs | data_0_10_1257811200 | table | postgres
 testNs | distinct             | table | postgres
 testNs | local_distinct       | table | postgres
 testNs | master               | table | postgres
 testNs | partition_0_10       | table | postgres
(6 rows)

SELECT *
FROM "testNs".cluster;
        time         | device_id | series_0 | series_1 | series_2 | series_bool 
---------------------+-----------+----------+----------+----------+-------------
 1257894000000000000 | dev1      |      1.5 |        1 |        2 | t
 1257894000000000000 | dev1      |      1.5 |        2 |          | 
 1257894000000001000 | dev1      |      2.5 |        3 |          | 
 1257894001000000000 | dev1      |      3.5 |        4 |          | 
 1257897600000000000 | dev1      |      4.5 |        5 |          | f
 1257894002000000000 | dev1      |      2.5 |        3 |          | 
 1257987600000000000 | dev1      |      1.5 |        1 |          | 
 1257987600000000000 | dev1      |      1.5 |        2 |          | 
 1257894000000000000 | dev2      |      1.5 |        1 |          | 
 1257894000000000000 | dev2      |      1.5 |        2 |          | 
(10 rows)

SELECT *
FROM "testNs".distinct;
   field   | value |  last_time_approx   
-----------+-------+---------------------
 device_id | dev1  | 1258074000000000000
 device_id | dev2  | 1257894000000000000
(2 rows)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs'));
                                                      json                                                       
-----------------------------------------------------------------------------------------------------------------
 {"time":1257987600000000000,"device_id":"dev1","series_0":1.5,"series_1":2,"series_2":null,"series_bool":null}
 {"time":1257987600000000000,"device_id":"dev1","series_0":1.5,"series_1":1,"series_2":null,"series_bool":null}
 {"time":1257897600000000000,"device_id":"dev1","series_0":4.5,"series_1":5,"series_2":null,"series_bool":false}
 {"time":1257894002000000000,"device_id":"dev1","series_0":2.5,"series_1":3,"series_2":null,"series_bool":null}
 {"time":1257894001000000000,"device_id":"dev1","series_0":3.5,"series_1":4,"series_2":null,"series_bool":null}
 {"time":1257894000000001000,"device_id":"dev1","series_0":2.5,"series_1":3,"series_2":null,"series_bool":null}
 {"time":1257894000000000000,"device_id":"dev1","series_0":1.5,"series_1":1,"series_2":2,"series_bool":true}
 {"time":1257894000000000000,"device_id":"dev2","series_0":1.5,"series_1":1,"series_2":null,"series_bool":null}
 {"time":1257894000000000000,"device_id":"dev1","series_0":1.5,"series_1":2,"series_2":null,"series_bool":null}
 {"time":1257894000000000000,"device_id":"dev2","series_0":1.5,"series_1":2,"series_2":null,"series_bool":null}
(10 rows)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate=> new_aggregate(10e9::BIGINT)
                     ));
                       json                        
---------------------------------------------------
 {"time":1257987600000000000,"sum(series_0)":3}
 {"time":1257897600000000000,"sum(series_0)":4.5}
 {"time":1257894000000000000,"sum(series_0)":14.5}
(3 rows)

